<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Introduction to Genetic Epidemiology - Schedule</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../scripts/conda.html" rel="next">
<link href="../index.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Schedule</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Introduction to Genetic Epidemiology</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/AndrewsLabUCSF/IntroGeneticEpi" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../index.html" class="sidebar-item-text sidebar-link">Introduction</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/schedule.html" class="sidebar-item-text sidebar-link active">Schedule</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/conda.html" class="sidebar-item-text sidebar-link">Conda Env</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/habshd.html" class="sidebar-item-text sidebar-link">HABS-HD</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/summarystats.html" class="sidebar-item-text sidebar-link">Summary Statistics</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Genome-wide Association Studies</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/gwas_qc.html" class="sidebar-item-text sidebar-link">GWAS QC</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/gwas.html" class="sidebar-item-text sidebar-link">GWAS</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/gwas_ss.html" class="sidebar-item-text sidebar-link">GWAS-SS</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/ancestry.html" class="sidebar-item-text sidebar-link">Genetic Ancestry</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/heritability.html" class="sidebar-item-text sidebar-link">Heritability</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/genetic_corr.html" class="sidebar-item-text sidebar-link">Genetic Correlations</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">Polygenic Risk Scores</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/prs.html" class="sidebar-item-text sidebar-link">Polygenic Risk Scores</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">Mendelian Randomization</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/mr.html" class="sidebar-item-text sidebar-link">Mendelian Randomization</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">Appendices</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../scripts/references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#week-1-0422---0426" id="toc-week-1-0422---0426" class="nav-link active" data-scroll-target="#week-1-0422---0426">Week 1 04/22 - 04/26</a>
  <ul class="collapse">
  <li><a href="#monday-0422" id="toc-monday-0422" class="nav-link" data-scroll-target="#monday-0422">Monday 04/22</a></li>
  <li><a href="#wensday-0424" id="toc-wensday-0424" class="nav-link" data-scroll-target="#wensday-0424">Wensday 04/24</a></li>
  <li><a href="#friday-0426" id="toc-friday-0426" class="nav-link" data-scroll-target="#friday-0426">Friday 04/26</a></li>
  </ul></li>
  <li><a href="#week-2-0429---0503" id="toc-week-2-0429---0503" class="nav-link" data-scroll-target="#week-2-0429---0503">Week 2 04/29 - 05/03</a>
  <ul class="collapse">
  <li><a href="#monday-0429" id="toc-monday-0429" class="nav-link" data-scroll-target="#monday-0429">Monday 04/29</a></li>
  <li><a href="#wensday-0501" id="toc-wensday-0501" class="nav-link" data-scroll-target="#wensday-0501">Wensday 05/01</a></li>
  <li><a href="#friday-0503" id="toc-friday-0503" class="nav-link" data-scroll-target="#friday-0503">Friday 05/03</a></li>
  </ul></li>
  <li><a href="#week-3-0506---0510" id="toc-week-3-0506---0510" class="nav-link" data-scroll-target="#week-3-0506---0510">Week 3 05/06 - 05/10</a>
  <ul class="collapse">
  <li><a href="#monday-0506" id="toc-monday-0506" class="nav-link" data-scroll-target="#monday-0506">Monday 05/06</a></li>
  <li><a href="#wensday-0508" id="toc-wensday-0508" class="nav-link" data-scroll-target="#wensday-0508">Wensday 05/08</a></li>
  <li><a href="#friday-0510" id="toc-friday-0510" class="nav-link" data-scroll-target="#friday-0510">Friday 05/10</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Schedule</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Welcome to this BMI Mini-Course on an Methods in Genetic Epidemiology. The course structure is divided into three key methodologies: genome-wide association studies, polygenic risk scores, and Mendelian randomization. Each trainee will apply these methods to a trait of their choosing and, at the course’s conclusion, present their findings to the class.</p>
<section id="week-1-0422---0426" class="level2">
<h2 class="anchored" data-anchor-id="week-1-0422---0426">Week 1 04/22 - 04/26</h2>
<section id="monday-0422" class="level3">
<h3 class="anchored" data-anchor-id="monday-0422">Monday 04/22</h3>
<p><strong>Alzheimer’s disease</strong></p>
<p>In this mini-course, we will frequently reference Alzheimer’s disease to illustrate various genetic epidemiology methods. This session aims to introduce Alzheimer’s disease and explore significant findings related to its genetic architecture.</p>
<p><strong>Readings</strong></p>
<ul>
<li>Knopman, D. S. et al.&nbsp;Alzheimer disease. <a href="https://doi.org/10.1038/s41572-021-00269-y">Nat Rev Dis Primers 7, 33 (2021)</a>.</li>
<li>Alzheimer’s Association. 2023 Alzheimer’s disease facts and figures. <a href="https://doi.org/10.1002/alz.13016">Alzheimer’s Dementia (2023)</a>.</li>
<li>Kornblith, E. et al.&nbsp;Association of Race and Ethnicity With Incidence of Dementia Among Older Adults. <a href="https://doi.org/10.1001/jama.2022.3550">Jama 327, 1488–1495 (2022)</a>.</li>
<li>Andrews, S. J. et al.&nbsp;The complex genetic architecture of Alzheimer’s disease: novel insights and future directions. <a href="https://doi.org/10.1016/j.ebiom.2023.104511">eBioMedicine 90, 104511 (2023)</a>.</li>
<li>Andrews, S. J., Fulton-Howard, B. &amp; Goate, A. Interpretation of risk loci from genome-wide association studies of Alzheimer’s disease. <a href="https://doi.org/10.1016/s1474-4422(19)30435-1">Lancet Neurology 19, 326–335 (2020)</a>.</li>
</ul>
</section>
<section id="wensday-0424" class="level3">
<h3 class="anchored" data-anchor-id="wensday-0424">Wensday 04/24</h3>
<p><strong>Genome-Wide Association Studies</strong></p>
<p>Genome-Wide Association Studies (GWAS) are foundational to various genetic analysis methodologies. In this session, we will delve into what a GWAS entails, the process of conducting one, and then engage in a hands-on exercise to carry out our own GWAS.</p>
<p><strong>Readings</strong></p>
<ul>
<li>Uffelmann, E. et al.&nbsp;Genome-wide association studies. <a href="https://www.nature.com/articles/s43586-021-00056-9">Nat Rev Methods Primers 1, 59 (2021)</a>.</li>
<li>Abdellaoui, A., Yengo, L., Verweij, K. J. H. &amp; Visscher, P. M. 15 years of GWAS discovery: Realizing the promise. <a href="https://doi.org/10.1016/j.ajhg.2022.12.011">Am J Hum Genetics (2023)</a></li>
<li>Marees, A. T. et al.&nbsp;A tutorial on conducting genome‐wide association studies: Quality control and statistical analysis. <a href="https://doi.org/10.1002/mpr.1608">Int J Method Psych 27, e1608 (2018)</a>.</li>
<li>MacArthur, J. A. L. et al.&nbsp;Workshop proceedings: GWAS summary statistics standards and sharing. <a href="https://pubmed.ncbi.nlm.nih.gov/36082306/">Cell Genom 1, 100004 (2021)</a>.</li>
</ul>
<p><strong>Tools</strong></p>
<ul>
<li><a href="https://www.cog-genomics.org/plink/">PLINK</a><span class="citation" data-cites="purcell2007"><sup><a href="references.html#ref-purcell2007" role="doc-biblioref">1</a></sup></span></li>
<li><a href="https://samtools.github.io/bcftools/howtos/index.html">BCFTools</a><span class="citation" data-cites="Danecek2021"><sup><a href="references.html#ref-Danecek2021" role="doc-biblioref">2</a></sup></span></li>
<li><a href="https://neurogenomics.github.io/MungeSumstats/">MungeSumStats</a><span class="citation" data-cites="Murphy2021"><sup><a href="references.html#ref-Murphy2021" role="doc-biblioref">3</a></sup></span></li>
</ul>
</section>
<section id="friday-0426" class="level3">
<h3 class="anchored" data-anchor-id="friday-0426">Friday 04/26</h3>
<p><strong>Genetic Ancestry</strong></p>
<p>Genetic ancestry explores the lineage and heritage inferred from our DNA, providing insights into population history and individual heritage. This session will introduce the concepts and methodologies used in determining genetic ancestry, emphasizing their importance in genetic epidemiology research.</p>
<p><strong>Readings</strong></p>
<ul>
<li>Lewis, A. C. F. et al.&nbsp;Getting genetic ancestry right for science and society. <a href="https://doi.org/10.1126/science.abm7530">Science 376, 250–252 (2022)</a>.</li>
<li>National Academies of Sciences, Engineering, and Medicine. 2023. Using Population Descriptors in Genetics and Genomics Research: A New Framework for an Evolving Field. <a href="https://doi.org/10.17226/26902">Washington, DC: The National Academies Press</a>.</li>
</ul>
<p><strong>Tools</strong></p>
<ul>
<li><a href="https://dalexander.github.io/admixture/">ADMIXTURE</a><span class="citation" data-cites="Alexander2009"><sup><a href="references.html#ref-Alexander2009" role="doc-biblioref">4</a></sup></span></li>
<li><a href="https://github.com/slowkoni/rfmix">RFmix</a><span class="citation" data-cites="Maples2013"><sup><a href="references.html#ref-Maples2013" role="doc-biblioref">5</a></sup></span></li>
</ul>
</section>
</section>
<section id="week-2-0429---0503" class="level2">
<h2 class="anchored" data-anchor-id="week-2-0429---0503">Week 2 04/29 - 05/03</h2>
<section id="monday-0429" class="level3">
<h3 class="anchored" data-anchor-id="monday-0429">Monday 04/29</h3>
<p><strong>Heritability &amp; Genetic Correlations</strong></p>
<p>Heritability quantifies the proportion of phenotype variance attributable to genetic factors, whereas genetic correlations assess the extent of shared genetic architecture between traits. In this session, we will concentrate on the tools utilized to estimate these metrics from GWAS summary statistics.</p>
<p><strong>Readings</strong></p>
<ul>
<li>Rheenen, W. van, Peyrot, W. J., Schork, A. J., Lee, S. H. &amp; Wray, N. R. Genetic correlations of polygenic disease traits: from theory to practice. <a href="https://doi.org/10.1038/s41576-019-0137-z">Nat Rev Genetics 20, 567–581 (2019)</a>.</li>
<li>Barry, C.-J. S. et al.&nbsp;How to estimate heritability: a guide for genetic epidemiologists. <a href="https://doi.org/10.1093/ije/dyac224">Int J Epidemiol (2022)</a></li>
</ul>
<p><strong>Tools</strong></p>
<ul>
<li><a href="https://github.com/bulik/ldsc">LDSC</a><span class="citation" data-cites="BulikSullivan2015"><sup><a href="references.html#ref-BulikSullivan2015" role="doc-biblioref">6</a></sup></span></li>
<li><a href="https://github.com/zhenin/HDL">HDL</a><span class="citation" data-cites="Ning2020"><sup><a href="references.html#ref-Ning2020" role="doc-biblioref">7</a></sup></span></li>
<li><a href="https://github.com/GenomicSEM/GenomicSEM">GenomicSEM</a><span class="citation" data-cites="Grotzinger2019"><sup><a href="references.html#ref-Grotzinger2019" role="doc-biblioref">8</a></sup></span></li>
</ul>
</section>
<section id="wensday-0501" class="level3">
<h3 class="anchored" data-anchor-id="wensday-0501">Wensday 05/01</h3>
<p><strong>Polygenic Risk Scores I</strong></p>
<p>Polygenic risk scores (PRS) measure an invididueals total genetic liability for a trait. This session will cover the process of constructing a PRS and assessing its performance in predicting the trait.</p>
<p><strong>Readings</strong></p>
<ul>
<li>Choi, S. W., Mak, T. S.-H. &amp; O’Reilly, P. F. Tutorial: a guide to performing polygenic risk score analyses. <a href="https://www.nature.com/articles/s41596-020-0353-1">Nat Protoc 15, 2759–2772 (2020)</a>.</li>
<li>Wand, H. et al.&nbsp;Improving reporting standards for polygenic scores in risk prediction studies. <a href="https://www.nature.com/articles/s41586-021-03243-6">Nature 591, 211–219 (2021)</a>.</li>
<li>Lennon, N. J. et al.&nbsp;Selection, optimization and validation of ten chronic disease polygenic risk scores for clinical implementation in diverse US populations. <a href="doi:10.1038/s41591-024-02796-z">Nat. Med. 1–8 (2024)</a></li>
</ul>
<p><strong>Tools</strong></p>
<ul>
<li><a href="https://choishingwan.github.io/PRSice/">PRSice2</a><span class="citation" data-cites="Choi2019"><sup><a href="references.html#ref-Choi2019" role="doc-biblioref">9</a></sup></span></li>
<li><a href="https://choishingwan.github.io/PRSice/quick_start_prset/">PRSet</a><span class="citation" data-cites="Choi2023"><sup><a href="references.html#ref-Choi2023" role="doc-biblioref">10</a></sup></span></li>
</ul>
</section>
<section id="friday-0503" class="level3">
<h3 class="anchored" data-anchor-id="friday-0503">Friday 05/03</h3>
<p><strong>Polygenic Risk Scores II</strong></p>
<p>The accuracy of polygenic risk scores (PRS) diminishes as the genetic distance from the training population increases. This session will explore cross-ancestry PRS methods designed to enhance PRS accuracy across diverse populations.</p>
<p><strong>Readings</strong></p>
<ul>
<li>Kachuri, L. et al.&nbsp;Principles and methods for transferring polygenic risk scores across global populations. <a href="https://www.nature.com/articles/s41576-023-00637-2">Nat. Rev.&nbsp;Genet. 1–18 (2023) doi:10.1038/s41576-023-00637-2</a>.</li>
<li>Ding, Y. et al.&nbsp;Polygenic scoring accuracy varies across the genetic ancestry continuum. <a href="https://www.nature.com/articles/s41586-023-06079-4">Nature 618, 774–781 (2023)</a>.</li>
</ul>
<p><strong>Tools</strong></p>
<ul>
<li><a href="https://github.com/getian107/PRScsx">PRS-CSx</a><span class="citation" data-cites="Ruan2022"><sup><a href="references.html#ref-Ruan2022" role="doc-biblioref">11</a></sup></span></li>
</ul>
</section>
</section>
<section id="week-3-0506---0510" class="level2">
<h2 class="anchored" data-anchor-id="week-3-0506---0510">Week 3 05/06 - 05/10</h2>
<section id="monday-0506" class="level3">
<h3 class="anchored" data-anchor-id="monday-0506">Monday 05/06</h3>
<p><strong>Mendelian Randomization I</strong></p>
<p>Mendelian Randomization (MR) is a method employed to identify causal risk factors for diseases. This session will cover the fundamentals of MR and demonstrate how to execute a two-sample MR analysis.</p>
<p><strong>Readings</strong></p>
<ul>
<li>Sanderson, E. et al.&nbsp;Mendelian randomization. <a href="https://www.nature.com/articles/s43586-021-00092-5">Nat Rev Methods Primers 2, 6 (2022)</a>.</li>
<li>Davies, N. M., Holmes, M. V. &amp; Smith, G. D. Reading Mendelian randomisation studies: a guide, glossary, and checklist for clinicians. <a href="https://www.bmj.com/content/362/bmj.k601">BMJ 362, k601 (2017)</a>.</li>
<li>Hemani, G. et al.&nbsp;The MR-Base platform supports systematic causal inference across the human phenome. <a href="https://pubmed.ncbi.nlm.nih.gov/29846171/">Elife 7, e34408 (2018)</a>.</li>
</ul>
<p><strong>Tools</strong></p>
<ul>
<li><a href="https://mrcieu.github.io/TwoSampleMR/">TwoSampleMR</a><span class="citation" data-cites="Hemani2018"><sup><a href="references.html#ref-Hemani2018" role="doc-biblioref">12</a></sup></span></li>
</ul>
</section>
<section id="wensday-0508" class="level3">
<h3 class="anchored" data-anchor-id="wensday-0508">Wensday 05/08</h3>
<p><strong>Mendelian Randomization II</strong></p>
<p>A crucial aspect of Mendelian Randomization (MR) studies is assessing whether the causal associations derived from MR analyses remain valid despite potential violations of MR’s underlying assumptions. This session will focus on diagnostic and sensitivity analyses in MR, along with guidance on effectively reporting MR findings.</p>
<p><strong>Readings</strong></p>
<ul>
<li>Skrivankova, V. W. et al.&nbsp;Strengthening the Reporting of Observational Studies in Epidemiology Using Mendelian Randomization. <a href="https://jamanetwork.com/journals/jama/fullarticle/2785494">JAMA 326, 1614–1621 (2021)</a>.</li>
<li>Skrivankova, V. W. et al.&nbsp;Strengthening the reporting of observational studies in epidemiology using mendelian randomisation (STROBE-MR): explanation and elaboration. <a href="https://www.bmj.com/content/375/bmj.n2233">BMJ 375, n2233 (2021)</a>.</li>
</ul>
</section>
<section id="friday-0510" class="level3">
<h3 class="anchored" data-anchor-id="friday-0510">Friday 05/10</h3>
<p><strong>Trainee Presentations</strong></p>
<p>Upon concluding this mini-course, trainees will showcase the results of their analyses.</p>


<div id="refs" class="references csl-bib-body" role="doc-bibliography" style="display: none">
<div id="ref-purcell2007" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Purcell S, Neale B, Todd-Brown K, et al. <a href="https://doi.org/10.1086/519795"><span class="nocase">PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses</span></a>. <em>The American Journal of Human Genetics</em>. 2007;81(3):559-575.</div>
</div>
<div id="ref-Danecek2021" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Danecek P, Bonfield JK, Liddle J, et al. <a href="https://doi.org/10.1093/gigascience/giab008"><span class="nocase">Twelve years of SAMtools and BCFtools</span></a>. <em>GigaScience</em>. 2021;10(2):giab008.</div>
</div>
<div id="ref-Murphy2021" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Murphy AE, Schilder BM, Skene NG. <a href="https://doi.org/10.1093/bioinformatics/btab665"><span class="nocase">MungeSumstats: A Bioconductor package for the standardisation and quality control of many GWAS summary statistics</span></a>. <em>Bioinformatics</em>. 2021;37(23):btab665-.</div>
</div>
<div id="ref-Alexander2009" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Alexander DH, Novembre J, Lange K. <a href="https://doi.org/10.1101/gr.094052.109"><span class="nocase">Fast model-based estimation of ancestry in unrelated individuals</span></a>. <em>Genome Research</em>. 2009;19(9):1655-1664.</div>
</div>
<div id="ref-Maples2013" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Maples BK, Gravel S, Kenny EE, Bustamante CD. <a href="https://doi.org/10.1016/j.ajhg.2013.06.020"><span class="nocase">RFMix: A Discriminative Modeling Approach for Rapid and Robust Local-Ancestry Inference</span></a>. <em>The American Journal of Human Genetics</em>. 2013;93(2):278-288.</div>
</div>
<div id="ref-BulikSullivan2015" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Bulik-Sullivan B, Finucane HK, Anttila V, et al. <a href="https://doi.org/10.1038/ng.3406"><span class="nocase">An atlas of genetic correlations across human diseases and traits</span></a>. <em>Nature Genetics</em>. 2015;47(11):1236-1241.</div>
</div>
<div id="ref-Ning2020" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Ning Z, Pawitan Y, Shen X. <a href="https://doi.org/10.1038/s41588-020-0653-y"><span class="nocase">High-definition likelihood inference of genetic correlations across human complex traits</span></a>. <em>Nature Genetics</em>. 2020;52(8):859-864.</div>
</div>
<div id="ref-Grotzinger2019" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Grotzinger AD, Rhemtulla M, Vlaming R de, et al. <a href="https://doi.org/10.1038/s41562-019-0566-x"><span class="nocase">Genomic structural equation modelling provides insights into the multivariate genetic architecture of complex traits.</span></a> <em>Nature human behaviour</em>. 2019;3(5):513-525.</div>
</div>
<div id="ref-Choi2019" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Choi SW, O’Reilly PF. <a href="https://doi.org/10.1093/gigascience/giz082"><span class="nocase">PRSice-2: Polygenic Risk Score software for biobank-scale data</span></a>. <em>GigaScience</em>. 2019;8(7).</div>
</div>
<div id="ref-Choi2023" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Choi SW, García-González J, Ruan Y, et al. <a href="https://doi.org/10.1371/journal.pgen.1010624"><span class="nocase">PRSet: Pathway-based polygenic risk score analyses and software</span></a>. <em>PLOS Genetics</em>. 2023;19(2):e1010624.</div>
</div>
<div id="ref-Ruan2022" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Ruan Y, Lin YF, Feng YCA, et al. <a href="https://doi.org/10.1038/s41588-022-01054-7"><span class="nocase">Improving polygenic prediction in ancestrally diverse populations</span></a>. <em>Nature Genetics</em>. 2022;54(5):573-580.</div>
</div>
<div id="ref-Hemani2018" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Hemani G, Zheng J, Elsworth B, et al. <a href="https://doi.org/10.7554/elife.34408"><span class="nocase">The MR-Base platform supports systematic causal inference across the human phenome</span></a>. <em>eLife</em>. 2018;7:e34408.</div>
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../index.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Introduction</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../scripts/conda.html" class="pagination-link">
        <span class="nav-page-text">Conda Env</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>